include/io.tcc

```diff
@@ -1,25 +1,61 @@
-//! \brief Write output file for point
-//! \details Get stress vector of vertex coordinates, and number of vertices
+//! Constructor with json input file
+//! Get mesh_filename and output_directory
+//! \param[in] json input file name
+template <unsigned Tdim>
+IO<Tdim>::IO(const std::string& file_directory, const std::string& json_file)
+    : file_directory_{file_directory} {
+
+  json_filename_ = file_directory_ + json_file;
+
+  //! Check if json file is present
+  std::ifstream inputFile(json_filename_);
+  inputFile.exceptions(std::ifstream::badbit);
+
+  // //! Store json object as private variable
+  // //! Read json file and store to private variables
+  inputFile >> json_file_;
+  mesh_filename_ =
+      file_directory_ + json_file_["mesh_file"].template get<std::string>();
+
+  //! Check if mesh file is present
+  std::ifstream meshfile;
+  meshfile.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+  try {
+    meshfile.open(mesh_filename_);
+  } catch (const std::ifstream::failure& except) {
+    std::cerr << "Exception opening/reading mesh file";
+  }
+  meshfile.close();
+
+  //! Material point and stresses output file
+  material_points_filename_ = file_directory_ + "material_points.txt";
+  stress_filename_ = file_directory_ + "initial_stresses.txt";
+}
+
+//! \brief output coordinates of material points
+//! \details Get vector of point coordinates
 //! \tparam Tdim dimension
 template <unsigned Tdim>
-void IO<Tdim>::write_material_points(
-    const std::vector<std::shared_ptr<Point<Tdim>>>& materialpoints) {
+void IO<Tdim>::write_point_coordinates(
+    const std::vector<Eigen::VectorXd>& coordinates) {
+
+  std::cout << "material_points will be stored in: " << file_directory_ << "\n";
```

Comment:
be consistent `\n`

Reply:
\n always after a statement, not before in this code